model/src/main/kotlin/licenses/ResolvedLicenseInfo.kt

```diff
@@ -73,6 +77,40 @@ data class ResolvedLicenseInfo(
             resolvedLicense.originalExpressions.map { it.expression }
         }.andOrNull()
 
+    /**
+     * Return the main license of a package (or project) as an [SpdxExpression], or null if there is no main license.
+     * The main license is the conjunction of the declared license of the package (or project) and the licenses detected
+     * in any of the configured [LicenseFilePatterns] matched against the root path of the package (or project).
+     */
+    fun mainLicense(): SpdxExpression? {
+        val matcher = PathLicenseMatcher(LicenseFilePatterns.getInstance())
+        val licensePaths = flatMap { resolvedLicense ->
+            resolvedLicense.locations.map { it.location.path }
+        }
+
+        val detectedLicenses = filterTo(mutableSetOf()) { resolvedLicense ->
+            resolvedLicense.locations.any {
+                val rootPath = (it.provenance as? RepositoryProvenance)?.vcsInfo?.path.orEmpty()
+
+                val applicableLicensePaths = matcher.getApplicableLicenseFilesForDirectories(
+                    licensePaths,
+                    listOf(rootPath)
+                )
+
+                val applicableLicenseFiles = applicableLicensePaths[rootPath].orEmpty()
+
+                it.location.path in applicableLicenseFiles
+            }
+        }
+
+        val declaredLicenses = filter(LicenseView.ONLY_DECLARED)
+        val mainLicenses = (detectedLicenses + declaredLicenses.licenses)
+            .flatMap { it.originalExpressions }
```

Comment:
I'm aware that the code existed before, but I wonder how this is supposed to work with licenses choices. 
In particular as we use the `originalExpressions` here. What do you think?

Reply:
Good question. I probably fell into the trap described [here](https://github.com/oss-review-toolkit/ort/issues/8013) again. If you agree, I'd like to postpone solving this when looking at https://github.com/oss-review-toolkit/ort/issues/10047 (and https://github.com/oss-review-toolkit/ort/issues/10082).

Reply:
So, if this instance of `ResolvedLicenseInfo` has been created by `.filterExcluded().applyChoices()` like call chain, then the choices should probably work. But, I have a hunch the `originalExpressions` does not have any choices applied. So, I was wondering whether `originalExpressions` breaks the choices and if there is an alternative way to map it, which doesn't break that.

Anyhow, I'm fine with leaving it up for later. Maybe a code comment could be valuable (next to `.flatMap { it.originalExpressions }`.